pkg/integrations/mongodb_exporter/logruskit.go

```diff
@@ -0,0 +1,103 @@
+package mongodb_exporter //nolint:golint
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/go-kit/log/level"
+
+	"github.com/go-kit/log"
+	"github.com/sirupsen/logrus"
+)
+
+// NewLogger transalates log.Logger to logrus.Logger
+func NewLogger(logger log.Logger, opts ...Option) *logrus.Logger {
+	o := output{}
+	for _, apply := range opts {
+		apply(&o)
+	}
+
+	l := logrus.New()
+	l.SetFormatter(formatter{})
+	l.SetOutput(output{l: logger})
+	return l
+}
+
+// Option Exposes logging options
+type Option func(*output)
+
+// WithTimestampFromLogrus sets tsFromLogrus logging option
+func WithTimestampFromLogrus() Option {
+	return func(o *output) {
+		o.tsFromLogrus = true
+	}
+}
+
+type output struct {
+	l            log.Logger
+	tsFromLogrus bool
+}
+
+func (o output) Write(data []byte) (n int, err error) {
+	var ll line
```

Comment:
Would love a bit more detail/comments on this translation function.

Reply:
The credit for this one is due to @colegam, he gave me this hint to consume the exporter without needing to change it`s Log signature.